perf/src/sigverify.rs

```diff
@@ -170,28 +169,51 @@ fn do_get_packet_offsets(
         .and_then(|v| v.checked_add(sig_size))
         .ok_or(PacketError::InvalidLen)?;
 
-    let msg_start_offset_plus_one = msg_start_offset
+    // Determine the start of the message header by checking the message prefix bit.
+    let msg_header_offset = {
+        let msg_start_offset_plus_one = msg_start_offset
+            .checked_add(1)
+            .ok_or(PacketError::InvalidLen)?;
+
+        // Packet should have data for prefix bit
+        if msg_start_offset_plus_one <= packet.meta.size {
+            return Err(PacketError::InvalidSignatureLen);
+        }
+
+        // next byte indicates if the transaction is versioned. If the top bit is set, the
+        // remaining bits encode a version number and the message header starts from the
+        // following byte. If the top bit is not set, this byte is the first byte of the
+        // message header.
+        let message_prefix = packet.data[msg_start_offset];
+        if message_prefix == MESSAGE_VERSION_PREFIX {
```

Comment:
```suggestion
        if message_prefix & MESSAGE_VERSION_PREFIX != 0 {
```
Otherwise v1 will fail, right?

Reply:
Yeah, but there is no v1 yet so it's fine to fail it